### PR TITLE
Add ragel 6.6 recipe.

### PR DESCRIPTION
**Pull request by rolando**
Ragel is a state machine compiler. This particular version aims to have
ragel 6.6 which is the version required for another library Squash.

Once this recipe is merged we plan to update it to newer ragel releases.

The maintainers list is open.

### CODE REVIEW

File: recipes/ragel/build.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+set -ex
+
+./configure --prefix=$PREFIX
+
+make -j${CPU_COUNT} VERBOSE=1
+make -j${CPU_COUNT} install
```
Comment (ocefpaf) [inline recipes/ragel/build.sh]: Is there a `make check` we can run here?

File: recipes/ragel/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{version }}.tar.gz
+  url: https://www.colm.net/files/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  {{ hash_type }}: {{ hash_val }}
+  patches:
+    # Patch shamelessly taken from MacPorts.
```
Comment (ocefpaf) [inline recipes/ragel/meta.yaml]: Please add a patch description or a link explaining why the need for it.

File: recipes/ragel/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    - patches/cxx_syntax.patch
+
+build:
+  number: 0
+  # TODO: Try to build on windows.
```
Comment (ocefpaf) [inline recipes/ragel/meta.yaml]: Please remove the comment and create an issue in the feedstock once that is created. (Comments are usually lost, issue can be tracked.)

File: recipes/ragel/build.sh (hunk 1 above)
Comment (rolando) [inline recipes/ragel/build.sh]: Yes, I added the `make check` call.

File: recipes/ragel/meta.yaml (hunk 2 above)
Comment (rolando) [inline recipes/ragel/meta.yaml]: I added a short description and link.

File: recipes/ragel/meta.yaml (hunk 3 above)
Comment (rolando) [inline recipes/ragel/meta.yaml]: Fixed.

**Pull request merged**